cinema/views.py

```diff
@@ -1,45 +1,92 @@
-from rest_framework.decorators import api_view
 from rest_framework.response import Response
-from rest_framework import status
+from rest_framework import status, generics, mixins, viewsets
 
 from django.shortcuts import get_object_or_404
+from rest_framework.views import APIView
 
-from cinema.models import Movie
-from cinema.serializers import MovieSerializer
+from cinema.models import Movie, Genre, Actor, CinemaHall
+from cinema.serializers import (
+    MovieSerializer,
+    GenreSerializer,
+    ActorSerializer,
+    CinemaHallSerializer
+)
 
 
-@api_view(["GET", "POST"])
-def movie_list(request):
-    if request.method == "GET":
-        movies = Movie.objects.all()
-        serializer = MovieSerializer(movies, many=True)
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = (Movie.objects.all().
+                prefetch_related("actors").
+                prefetch_related("genres"))
+    serializer_class = MovieSerializer
+
+
+class GenreList(APIView):
```

Comment:
Consider using `generics.ListCreateAPIView` for `GenreList` to reduce boilerplate code, as it already provides the `get` and `post` methods.